thavalon-webapp/src/components/gameComponents/playerBoard.tsx

```diff
@@ -1,5 +1,177 @@
+    const [tabbedOutPlayers, setTabbedOutPlayers] = useState(new Set<string>());
+
+    /**
+     * Generic message handler for all messages from the server
+     * @param message The InboundMessage from the server.
+     */
+    function handleMessage(message: InboundMessage): void {
+        switch (message.messageType) {
+            case InboundMessageType.Snapshot:
+                const snapshot = message.data as Snapshot
+                handleGameMessage(snapshot.log[0]);
+                break;
+            case InboundMessageType.PlayerFocusChange:
+                const { displayName, isTabbedOut } = message.data as PlayerFocusChangeMessage;
+                playerFocusChanged(displayName, isTabbedOut);
+                break;
+            case InboundMessageType.GameMessage:
+                handleGameMessage(message.data as GameMessage);
+                break;
+        }
+    }
+
+    /**
+     * GameMessage specific message handler. This is needed because the GameMessage
+     * has internal types to parse.
+     * @param message The GameMessage from the server
+     */
+    function handleGameMessage(message: GameMessage): void {
+        switch (message.messageType) {
+            case GameMessageType.ProposalOrder:
+                setPlayerList(message.data as string[]);
+                break;
+        }
+    }
+
+    /**
+     * Updates the selected player list with a new name.
+     * @param name The player name to select.
+     */
+    function updateSelectedPlayers(name: string): void {
+        updateSet(selectedPlayers, name, setSelectedPlayers);
+    }
+
+    /**
+     * Sends a message to the server that the player is tabbed in or out.
+     */
+    function sendPlayerVisibilityChange(): void {
+        const connection = GameSocket.getInstance();
+        const isTabbedOut = document.visibilityState === "hidden" ? true : false;
+        const message = { messageType: OutboundMessageType.PlayerFocusChange, data: isTabbedOut };
+        connection.sendMessage(message);
+    }
+
+    /**
+     * Updates the tabbed out player list with a new player and visibility
+     * @param player The player whose focus has changed
+     * @param visibility The new visibility for that player
+     */
+    function playerFocusChanged(player: string, isTabbedOut: boolean): void {
+        const tempSet = new Set(tabbedOutPlayers.values());
+        if (isTabbedOut) {
+            tempSet.add(player);
+        } else {
+            tempSet.delete(player);
+        }
+        setTabbedOutPlayers(tempSet);
+    }
+
+    /**
+     * Helper function to update a set.
+     * @param setToUpdate The set to update
+     * @param valueToToggle The value in the set to toggle
+     * @param reactSetter The React setter for the state to update
+     */
+    function updateSet<T>(setToUpdate: Set<T>, valueToToggle: T, reactSetter: React.Dispatch<React.SetStateAction<Set<T>>>): void {
+        const tempSet = new Set<T>(setToUpdate.values());
```

Comment:
can you add a comment clarifying why we need to make a new set before updating state?